### ISSUE TITLE
VSCode debugging is macOS-specific

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#1301]
In `.vscode/launch.json`:

>  "runtimeExecutable": "${workspaceRoot}/dist/GitHub Desktop-dev-darwin-x64/GitHub Desktop-dev.app/Contents/MacOS/GitHub Desktop-dev",

I guess we need to make these platform-specific

**Comment by shiftkey** [comment]
Giving this to @iAmWillShepherd to do some digging into what's changed since we last knew it was working

**Comment by shiftkey** [comment]
An interesting tidbit - `electron-forge` still supports debugging from VSCode using the one config:

```js
{
  "type": "node",
  "request": "launch",
  "name": "Electron Main",
  "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron-forge-vscode-nix",
  "windows": {
    "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron-forge-vscode-win.cmd"
  },
  // runtimeArgs will be passed directly to your Electron application 
  "runtimeArgs": [
    "foo",
    "bar"
  ],
  "cwd": "${workspaceRoot}"
}
```

The `windows` section is something we can do too to simplify things, but I'll also confirm this is working fine with the latest bits for VSCode.

**Comment by iAmWillShepherd** [comment]
How is debugging working for you at the moment?

**Comment by shiftkey** [comment]
It doesn't seem to be working since we updated to webpack 2, as the config returned has changed significantly:

```
$ npm run debug

> @ debug /Users/shiftkey/src/desktop/desktop
> cross-env NODE_ENV=development node script/debug

/Users/shiftkey/src/desktop/desktop/script/debug:13
  publicPath: config.output.publicPath,
                           ^

TypeError: Cannot read property 'publicPath' of undefined
    at Object.<anonymous> (/Users/shiftkey/src/desktop/desktop/script/debug:13:28)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:151:9)
    at bootstrap_node.js:542:3
```

Applying [this version](https://github.com/desktop/desktop/blob/17b39d7f2b1e0515f506ee2425a50ab38b34a153/script/debug) makes the error go away. I think the switch to `parallel-webpack` has changed other things that we relied on for this script, as running the Renderer as mentioned [here](https://github.com/desktop/desktop/blob/master/docs/contributing/tooling.md#visual-studio-code-1) doesn't launch anything 🤔

**Comment by iAmWillShepherd** [comment]
Yea I noticed that about WebPack too. I was not able to figure out how to get the debugger to attach to the renderer. I think maintaining debugging support in VSCode may be too fragile to be worthwhile. Debugging support actually stopped for me when we upgraded node and since then chrome dev tools, electron, and web pack have been updated.

If you really wanna get this locked in (and maybe release a package), I'd be happy to pair with you while in SF.
